mass_mailing_partner/views/mailing_contact_view.xml

```diff
@@ -30,31 +30,37 @@
                 <attribute
                     name="attrs"
                 >{'readonly': [('partner_id', '!=', False)]}</attribute>
+                <attribute name="force_save">1</attribute>
             </field>
             <field name="email" position="attributes">
                 <attribute
                     name="attrs"
-                >{'readonly': [('partner_id', '!=', False)]}</attribute>
+                >{'readonly': [('partner_id', '!=', False), ('email', '!=', False)]}</attribute>
```

Comment:
Then AFAIK, the changes of this file are the only one affected. The rest is non needed refactoring.